MAINTAINERS.md

```diff
@@ -0,0 +1,77 @@
+# Guide for maintainers of LNDK
+
+## Release process
+
+We prefer having the entire lifecycle of the release process happen within GitHub actions so that there are no "loose ends"
+or inconsistencies between releases based on local environments.
+
+### Prerequisites
+
+```shell
+cargo install --locked cargo-dist cargo-release
+```
+
+### Use of `cargo-dist` in this project
+
+This project uses [cargo-dist] for automating the release process as much as possible.
+Release builds happen via GitHub Actions and the workflow can be seen [here](.github/workflows/release.yml). The file is generated
+by `cargo-dist` and should generally not be modified manually, but we've included an intermediate signing step with cosign, until
+`cargo-dist` supports it upstream at some stage.
+
+### Setting up signing of release artifacts
+
+We use [Sigstore]'s [Cosign] utility to have GitHub Actions sign a release `checksum.txt` file with a key generated as a GitHub Secret
+so that only GitHub Actions can sign.
+
+The key is generated by the lead maintainer (carlaKC) with the `cosign` CLI as follows:
+
+```shell
+cosign generate-key-pair github://carlaKC/lndk
+````
+
+NOTE:
+  * A password for the key MUST be provided.
+  * You must have a GitHub Personal Access Token in the `GITHUB_TOKEN` environment variable locally.
+    A [fine-grained token] with the "Secrets" permission under "Repository permissions"
+    set to "Read and write" access is sufficient. The token can be revoked after generating the key-pair.
+
+An output similar to below indicates a key was successfully generated:
+
+```
+Password written to COSIGN_PASSWORD github actions secret
+Private key written to COSIGN_PRIVATE_KEY github actions secret
+Public key written to COSIGN_PUBLIC_KEY github actions secret
+Public key also written to cosign.pub
+```
+The `cosign.pub` should be commited to the project's root along with its PGP signature (`cosign.pub.asc`) manually signed by carlaKC.
+This allows users to verify the key is authentic. Unfortunately, with a fulle GitHub CI actions process, keyless signing with GitHub
```

Comment:
fulle/full

Reply:
Thanks. Probably also not the right word lol. More like "process that exclusively runs on GitHub Action"